src/Generics/Deriving/Default.hs

```diff
@@ -0,0 +1,248 @@
+-- |
+-- Module      : Generics.Deriving.Default
+-- Description : Default implementations of generic classes
+-- License     : BSD-3-Clause
+--
+-- Maintainer  : generics@haskell.org
+-- Stability   : experimental
+-- Portability : non-portable
+--
+-- GHC 8.6 introduced the 'DerivingVia' language extension, which means
+-- an instance of a typeclass for a type can be derived from an instance
+-- for an isomorphic type.
+--
+-- There are a number of default instantiations for the classes to be
+-- derived via generics in this package. This module exports the 'Default'
+-- newtype, which indicates that a type should use those default instances
+-- for its typeclass instances.
+--
+-- Use as follows:
+--
+-- @
+-- data MyType = …
+--  deriving (Generic)
+--  deriving (Eq) via (Default MyType)
+--
+-- deriving via (Default MyType) instance Show MyType
+-- @
+
+{-# LANGUAGE CPP #-}
+#if __GLASGOW_HASKELL__ >= 701
+{-# LANGUAGE DefaultSignatures #-}
+{-# LANGUAGE Safe #-}
+#endif
+#if __GLASGOW_HASKELL__ >= 706
+{-# LANGUAGE InstanceSigs #-}
+#endif
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Generics.Deriving.Default
+  ( Default(..)
+  , Default1(..)
+  ) where
+
+import Data.Foldable
+
+import Generics.Deriving.Base
+import Generics.Deriving.Copoint
+import Generics.Deriving.Enum
+import Generics.Deriving.Eq
+import Generics.Deriving.Foldable
+import Generics.Deriving.Functor
+import Generics.Deriving.Monoid
+import Generics.Deriving.Semigroup
+import Generics.Deriving.Show
+import Generics.Deriving.Traversable
+import Generics.Deriving.Uniplate
+
+newtype Default a = Default { unDefault :: a }
+
+newtype Default1 f a = Default1 { unDefault1 :: f a }
+
+--------------------------------------------------------------------------------
+-- Eq
+--------------------------------------------------------------------------------
+
+instance (Generic a, GEq' (Rep a)) => Eq (Default a) where
```

Comment:
As I mentioned in https://github.com/dreixel/generic-deriving/issues/58#issuecomment-448391865, I wouldn't bother defining instances for classes that aren't defined in this library. In other words, drop the `Eq` instance, but keep the `GEq` instance. (And similarly for the other instances below these.)